resolver/resolver.cc

```diff
@@ -121,7 +121,8 @@ class ResolveConstantsWalk {
         core::SymbolRef klass;
         core::FileRef file;
 
-        bool isSuperclass; // true if superclass, false for mixin
+        enum Kind { superclass, mixin, requiredAncestor };
```

Comment:
Can you use the same case that you'd use for class names here? Also, use `enum class` in C++. It's a lot better than `enum` for a handful of reasons.
```suggestion
        enum class Kind { Superclass, Mixin, RequiredAncestor };
```